### PR TITLE
Various change to remove variables when unnecessary and simplify the calculator and other classes

### PR DESCRIPTION
**Pull request by greenscientist**
(based on previous PR)

### CODE REVIEW

**Comment by tahini** [comment]
Please rebase on latest v2c or wait until the other PR gets in before rebasing?

**Comment by greenscientist** [comment]
Yes, I'm waiting for the JourneyStep PR to be done before rebasing

**Comment by greenscientist** [comment]
(Did rebase on the current journeystep, make it easier to see which commits are only in this PR)

File: tests/connection_scan_algorithm/csa_result_to_response_test.hpp [hunk 1]
```diff
@@ -29,12 +35,22 @@ class ResultToResponseFixtureTest : public ::testing::Test
...
     inline static const std::string unboardingNodeUuid = "aaaaaaaa-6666-cccc-dddd-eeeeeeffffff";
-    inline static TrRouting::Point boardingNode = TrRouting::Point(45.526, -73.59);
-    inline static TrRouting::Point unboardingNode = TrRouting::Point(45.53, -73.61);
+    inline static TrRouting::Point boardingNodePoint = TrRouting::Point(45.526, -73.59);
+    inline static TrRouting::Point unboardingNodePoint = TrRouting::Point(45.53, -73.61);
 
     std::unique_ptr<TrRouting::Scenario> scenario;
     std::unique_ptr<TrRouting::RouteParameters> testParameters;
 
+  std::unique_ptr<TrRouting::Mode> mode;
```
Comment (tahini) [inline tests/connection_scan_algorithm/csa_result_to_response_test.hpp]: These lines are not aligned with the rest

File: include/routing_result.hpp [hunk 2]
```diff
@@ -166,31 +117,16 @@ namespace TrRouting
...
       int _arrivalTime,
       int _inVehicleTime,
       int _inVehicleDistanceMeters
-    ): TransitRoutingStep(result_step_type::UNBOARDING, _agencyUuid, _agencyAcronym, _agencyName,
-      _lineUuid, _lineShortname, _lineLongname, _pathUuid,
-      _modeName, _mode, _tripUuid, _legSequenceInTrip,
-      _stopSequenceInTrip, _nodeUuid, _nodeCode, _nodeName,
-      _nodeCoordinates), arrivalTime(_arrivalTime), inVehicleTime(_inVehicleTime),
-      inVehicleDistanceMeters(_inVehicleDistanceMeters)
+                   ): TransitRoutingStep(result_step_type::UNBOARDING, _trip, _legSequenceInTrip, _stopSequenceInTrip, _node),
```
Comment (tahini) [inline include/routing_result.hpp]: It' s normal that the closing parenthesis is so far right?

File: connection_scan_algorithm/include/calculator.hpp [hunk 3]
```diff
@@ -59,12 +59,12 @@ namespace TrRouting
...
-    std::optional<std::tuple<int, std::reference_wrapper<const Node>>> reverseCalculation(RouteParameters &parameters); // best departure time, best access node
+    std::optional<std::tuple<int, std::reference_wrapper<const Node>>> forwardCalculation(RouteParameters &parameters, std::unordered_map<Node::uid_t, JourneyStep> & forwardEgressJourneysSteps); // best arrival time,   best egress node
+    std::optional<std::tuple<int, std::reference_wrapper<const Node>>> reverseCalculation(RouteParameters &parameters, std::unordered_map<Node::uid_t, JourneyStep> & reverseAccessJourneysSteps); // best departure time, best access node
     // TODO See calculate
-    std::unique_ptr<RoutingResult> forwardJourneyStep(RouteParameters &parameters, int bestArrivalTime, std::optional<std::reference_wrapper<const Node>> bestEgressNode);
+    std::unique_ptr<RoutingResult> forwardJourneyStep(RouteParameters &parameters, int bestArrivalTime, std::optional<std::reference_wrapper<const Node>> bestEgressNode, const std::unordered_map<Node::uid_t, JourneyStep> & forwardEgressJourneysSteps);
```
Comment (tahini) [inline connection_scan_algorithm/include/calculator.hpp:66]: Does the `const` before the `forwardEgressJourneySteps` imply that the content of the map cannot be modified in this function. Iiuc, this map will be filled in the forwardCalculation function? And used in the forwardJourneyStep? If so, please add a comment, that one method receives an empty map, the other expects it filled.

**Review by tahini — CHANGES_REQUESTED** [review]
Looks ok, except for some formatting comments and more comments needed (like always...)

File: connection_scan_algorithm/include/calculator.hpp (hunk 3 above)
Comment (greenscientist) [reply connection_scan_algorithm/include/calculator.hpp:66]: This is what a const imply. I would argue that a comment is not really needed to explicit that.

Comment (tahini) [reply connection_scan_algorithm/include/calculator.hpp:66]: I thought the const implied the value, ie the map pointer, not its content. Or is that only for simple types?

Comment (greenscientist) [reply connection_scan_algorithm/include/calculator.hpp:66]: You won't be able to add/remove elements to a const map. And you will only have access to const references for the elements.

File: include/routing_result.hpp (hunk 2 above)
Comment (greenscientist) [reply include/routing_result.hpp]: Yes, it's aligned with the opening one. But I can bring it back down.

File: connection_scan_algorithm/include/calculator.hpp (hunk 3 above)
Comment (greenscientist) [reply connection_scan_algorithm/include/calculator.hpp:66]: added comment to the forward/reverveCalculation to specify that the journeySteps there is a return value also.

Comment (tahini) [reply connection_scan_algorithm/include/calculator.hpp:66]: Thanks

File: include/routing_result.hpp (hunk 2 above)
Comment (tahini) [reply include/routing_result.hpp]: You align the parenthesis with the opening one, no matter how far down the line it is? Is that a C++ standard style? I'm used to languages where the formatting automatically aligns it by default to the right indent. Just asking...

**Review by tahini — APPROVED** [review]

**Comment by greenscientist** [comment]
I usually align the thing inside the parenthesis to the opening one yes. 
That's how my emacs does it and it's configured by default. I don't think there's a strong consensus on that tho

**Pull request merged**